pkg/skaffold/deploy/util.go

```diff
@@ -167,3 +168,8 @@ func ApplyDefaultRepo(globalConfig string, defaultRepo *string, tag string) (str
 
 	return newTag, nil
 }
+
+// ApplyRepoPrefix applies prefix to the given image tag
+func ApplyRepoPrefix(prefix string, image string) string {
+	return path.Join(prefix, image)
```

Comment:
Oh, interesting!  My interpretation of the user requests was to have the prefix replace all image name components except for the last.  That is with `--repo-prefix gcr.io/k8s-skaffold`:

1. foo &rarr; gcr.io/k8s-skaffold/foo 
2. gcr.io/gcp-dev-tools/foo &rarr; gcr.io/k8s-skaffold/foo

whereas with your implementation:

1. foo &rarr; gcr.io/k8s-skaffold/foo 
2. gcr.io/gcp-dev-tools/foo &rarr; gcr.io/k8s-skaffold/gcr.io/gcp-dev-tools/foo

We currently have special-casing logic in `--default-repo` for GCR so that that it produces the `gcr.io/k8s-skaffold/gcr.io/gcp-dev-tools/foo` result, and the user feedback for `--default-repo` was that people found it confusing to have the existing image components retained.

That said, my interpretation could lead to conflicts such as the following, and we'll need to check for:

- gcr.io/my-project/foo &rarr; gcr.io/k8s-skaffold/foo 
- gcr.io/gcp-dev-tools/foo &rarr; gcr.io/k8s-skaffold/foo

Reply:
i think what you are referring to is work done in #4150.
I did not get time to finish it.